src/crypto/internal/boring/goboringcrypto.h

```diff
@@ -19,218 +19,664 @@
+{
+	return EVP_MD_type(arg0);
+}
+
+static inline size_t
+_goboringcrypto_EVP_MD_size(const GO_EVP_MD* arg0)
+{
+	return EVP_MD_size(arg0);
+}
+
+
+#include <openssl/hmac.h>
+
+typedef HMAC_CTX GO_HMAC_CTX;
+static inline void
+_goboringcrypto_HMAC_CTX_init(GO_HMAC_CTX* arg0)
+{
+	HMAC_CTX_init(arg0);
+}
+
+static inline void
+_goboringcrypto_HMAC_CTX_cleanup(GO_HMAC_CTX* arg0)
+{
+	HMAC_CTX_cleanup(arg0);
+}
+
+static inline int
+_goboringcrypto_HMAC_Init(GO_HMAC_CTX* arg0, const void* arg1, int arg2, const GO_EVP_MD* arg3)
+{
+	return HMAC_Init(arg0, arg1, arg2, arg3);
+}
+
+static inline int
+_goboringcrypto_HMAC_Update(GO_HMAC_CTX* arg0, const uint8_t* arg1, size_t arg2)
+{
+	return HMAC_Update(arg0, arg1, arg2);
+}
+
+static inline int
+_goboringcrypto_HMAC_Final(GO_HMAC_CTX* arg0, uint8_t* arg1, unsigned int* arg2)
+{
+	return HMAC_Final(arg0, arg1, arg2);
+}
+
+static inline size_t
+_goboringcrypto_HMAC_size(const GO_HMAC_CTX* arg0)
+{
+	return HMAC_size(arg0);
+}
+
 int _goboringcrypto_HMAC_CTX_copy_ex(GO_HMAC_CTX *dest, const GO_HMAC_CTX *src);
 
-// #include <openssl/aes.h>
-typedef struct GO_AES_KEY { char data[244]; } GO_AES_KEY;
-int _goboringcrypto_AES_set_encrypt_key(const uint8_t*, unsigned int, GO_AES_KEY*);
-int _goboringcrypto_AES_set_decrypt_key(const uint8_t*, unsigned int, GO_AES_KEY*);
-void _goboringcrypto_AES_encrypt(const uint8_t*, uint8_t*, const GO_AES_KEY*);
-void _goboringcrypto_AES_decrypt(const uint8_t*, uint8_t*, const GO_AES_KEY*);
-void _goboringcrypto_AES_ctr128_encrypt(const uint8_t*, uint8_t*, size_t, const GO_AES_KEY*, uint8_t*, uint8_t*, unsigned int*);
+#include <openssl/aes.h>
+
+typedef AES_KEY GO_AES_KEY;
+static inline int
+_goboringcrypto_AES_set_encrypt_key(const uint8_t* arg0, unsigned int arg1, GO_AES_KEY* arg2)
+{
+	return AES_set_encrypt_key(arg0, arg1, arg2);
+}
+
+static inline int
+_goboringcrypto_AES_set_decrypt_key(const uint8_t* arg0, unsigned int arg1, GO_AES_KEY* arg2)
+{
+	return AES_set_decrypt_key(arg0, arg1, arg2);
+}
+
+static inline void
+_goboringcrypto_AES_encrypt(const uint8_t* arg0, uint8_t* arg1, const GO_AES_KEY* arg2)
+{
+	AES_encrypt(arg0, arg1, arg2);
+}
+
+static inline void
+_goboringcrypto_AES_decrypt(const uint8_t* arg0, uint8_t* arg1, const GO_AES_KEY* arg2)
+{
+	AES_decrypt(arg0, arg1, arg2);
+}
+
+static inline void
+_goboringcrypto_AES_ctr128_encrypt(const uint8_t* arg0, uint8_t* arg1, size_t arg2, const GO_AES_KEY* arg3, uint8_t* arg4, uint8_t* arg5, unsigned int* arg6)
+{
+	AES_ctr128_encrypt(arg0, arg1, arg2, arg3, arg4, arg5, arg6);
+}
+
 enum {
 	GO_AES_ENCRYPT = 1,
 	GO_AES_DECRYPT = 0
 };
-void _goboringcrypto_AES_cbc_encrypt(const uint8_t*, uint8_t*, size_t, const GO_AES_KEY*, uint8_t*, const int);
+static inline void
+_goboringcrypto_AES_cbc_encrypt(const uint8_t* arg0, uint8_t* arg1, size_t arg2, const GO_AES_KEY* arg3, uint8_t* arg4, const int arg5)
+{
+	AES_cbc_encrypt(arg0, arg1, arg2, arg3, arg4, arg5);
+}
+
```

Comment:
Using the direct AES_xxx_encrypt/decrypt calls means that the AES-NI support would not be used. I do not think this is the right way. How are these functions used in the Go code, are they directly used for any AES encryption/decryption? The right thing would be to use EVP_CIPHER_CTX for the GO_AES_KEY, I think.